engine/lib/user_settings.php

```diff
@@ -85,6 +85,10 @@ function _elgg_set_user_name() {
 	$name = strip_tags(get_input('name'));
 	$user_guid = get_input('guid');
 
+	if (!isset($name)) {
+		return;
```

Comment:
Or maybe `return true;` to indicate success in making no change. If the function caller expects true...